src/components/ZetkinQuery.tsx

```diff
@@ -0,0 +1,66 @@
+import { ErrorOutlined } from '@material-ui/icons';
+import { FormattedMessage } from 'react-intl';
+import { Box, CircularProgress, Typography } from '@material-ui/core';
+import { QueryObserverSuccessResult, UseQueryResult } from 'react-query';
+
+interface ZetkinQueryProps<G> {
+    children?: React.ReactNode | (({ query }: {query: QueryObserverSuccessResult<G>}) => React.ReactNode);
+    query: UseQueryResult<G>;
+    errorIndicator?: React.ReactNode;
+    loadingIndicator?: React.ReactNode;
+}
+
+
+function ZetkinQuery<G>({
+    children,
+    query,
+    loadingIndicator,
+    errorIndicator,
+}: ZetkinQueryProps<G>): JSX.Element {
+    if (query.isLoading) {
+        return (
+            <>
+                { loadingIndicator || (
+                    <Box
+                        display="flex"
+                        justifyContent="center"
+                        padding={ 3 }
+                        width="100%">
+                        <CircularProgress data-testid="loading-indicator" disableShrink />
+                    </Box>
+                ) }
+            </>
+        );
```

Comment:
Is the fragment really required? I would have written this code like this (but not sure it would pass linters et c):

```tsx
return loadingIndicator || (
  <Box
    display="flex"
    justifyContent="center"
    padding={ 3 }
    width="100%">
    <CircularProgress data-testid="loading-indicator" disableShrink />
  </Box>
);
```

Reply:
Imlemented that change